services/roomService/src/lib/CoveyTownController.ts

```diff
@@ -195,4 +207,17 @@ export default class CoveyTownController {
       nearbyPlayers.includes(listener.getAssociatedPlayer()),
     );
   }
+  
+  /**
+   * Informs the correct listener that a direct message has been sent
+   * @param directMessageId
+   * @returns the listener that is recieving the message being sent
+   */
+  getDirectMessageListener(directMessageId: string): CoveyTownListener {
+    const matchingPlayerID = directMessageId.split(':')[1];
+    const directMessageListener = this._listeners.filter(
+      listener => listener.getAssociatedPlayer().id === matchingPlayerID,
+    );
+    return directMessageListener[0];
```

Comment:
Wait, shouldn't you be returning all of them? The way the frontend is set up, every player (including the one who sent the message) involved in a MessageChain needs to get the message.